### PR TITLE
Code simplifications found by cppcheck

### PR DESCRIPTION
**Pull request by Sauermann**
They are based on:
- Boolean arithmetic simplifications
- setting variables that are not accessed
- constant variables

Updated 2022-11-21: Fix merge conflict

### CODE REVIEW

File: platform/macos/export/export_plugin.cpp [hunk 1]
```diff
@@ -724,21 +724,14 @@ Error EditorExportPlatformMacOS::_code_sign(const Ref<EditorExportPreset> &p_pre
 			args.push_back(p_path);
 
 			String str;
-			int exitcode = 0;
 
 			Error err = OS::get_singleton()->execute("codesign", args, &str, nullptr, true);
```
Comment (bruvzg) [inline platform/macos/export/export_plugin.cpp:728]:
```suggestion
			Error err = OS::get_singleton()->execute("codesign", args, &str, &exitcode, true);
```
And keep the rest of the code as is.

File: servers/rendering/renderer_rd/framebuffer_cache_rd.h [hunk 2]
```diff
@@ -253,8 +253,7 @@ class FramebufferCacheRD : public Object {
 	}
 
 	RID get_cache_multipass(const Vector<RID> &p_textures, const Vector<RD::FramebufferPass> &p_passes, uint32_t p_views = 1) {
-		uint32_t h = hash_murmur3_one_32(p_views);
-		h = hash_murmur3_one_32(p_textures.size());
+		uint32_t h = hash_murmur3_one_32(p_textures.size());
```
Comment (bruvzg) [inline servers/rendering/renderer_rd/framebuffer_cache_rd.h:256]: Probably should be:
```suggestion
		uint32_t h = hash_murmur3_one_32(p_views);
		h = hash_murmur3_one_32(p_textures.size(), h);
```

Comment (Sauermann) [reply servers/rendering/renderer_rd/framebuffer_cache_rd.h:256]: It looks like this should be investigated more thoroughly. Will remove it from this PR.

File: platform/macos/export/export_plugin.cpp (hunk 1 above)
Comment (Sauermann) [reply platform/macos/export/export_plugin.cpp]: Thanks for your comments. Done.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**